# Experiment configuration

t_activate: 15.0
t_final: 500.0
scenario_id: 2 # 1: pop-up, 2: reduced mu turn, 3: racing
s_begin_log: 180
s_pause_gazebo: []
#s_pause_gazebo: [180, 195, 210, 225, 240] # snapshots

# pop-up obstacle config
s_ego_at_popup:  50
s_obs_at_popup: 75
d_obs_at_popup: -1.75 

### track config 
track_name: "asta_zero" # todo pipe through at "simulation.cfg"
s_begin_mu_segments:  [0.0, 200.0]
mu_segment_values:  [1.0, 0.5]

# cc config
cc_vxref: 15
cc_dref: 0

# pcs refs
pcs_vxref: 1.0
pcs_dref: -1

### saarti algorithm config		

# Reference mode, 
# 0: minimize s
# 1: cc @ cc_vxref
# 2: maximize s
ref_mode: 1

# Algorithm mode, 
# 0: RTI
# 1: SAA-RTI 
sampling_augmentation: 0
Nd_rollout: 10 # multiples of 32 to maximize gpu utilization
Nvx_rollout: 15
vxub_rollout: 20.0

# Traction adaptation 
# 0: nominal value of mu from param mu_nominal and static tire force constraints (no pitch dynamics), static Cr
# 1: takes mu estimate from pathlocal struct and computes normal forces from pitch dynamics, Cr from mu and Fz
traction_adaptive: 0
mu_nominal: 1.0 # used only when traction_adaptive = 0

# optimization config
Wx:  [1000.0, 10000000.0, 1.0, 0.1, 1000000.0, 0.1]
WNx: [1000.0, 100000000.0, 1.0, 0.1, 1000000.0, 0.1]
Wu: [0.001, 0.1, 0.1]
Wslack: 100000000
